appcli/backup_manager/remote_strategy_factory.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Factory class for getting remote backup strategies.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+# standard libraries
+from datetime import datetime
+from pathlib import Path
+
+# local libraries
+from appcli.backup_manager.remote_strategy import AwsS3Strategy
+from appcli.logger import logger
+
+
+class RemoteStrategyFactory:
+    """
+    Factory for getting all remote strategies that match the config
+    """
+
+    @staticmethod
+    def get_strategy(backup_manager, key_file: Path):
+        strategies = {"S3": AwsS3Strategy}
+
+        backup_strategies = []
+
+        for backup in backup_manager.remote_strategy_config:
+            strategy_class = strategies.get(
+                backup["type"], lambda: "Invalid remote strategy"
+            )
+
+            if isinstance(strategy_class, str):
+                logger.error(
+                    f"No remote backup strategies found for type {backup['type']}"
+                )
+
+            strategy = strategy_class(backup, key_file)
+
+            if RemoteStrategyFactory.__frequency_check(strategy.frequency):
+                backup_strategies.append(strategy)
+
+        return backup_strategies
+
+    def __frequency_check(frequency):
+        """
+        Check if today matches the pseudo cron `frequency`.
+        Frequency format is `* * *`:
+        First `*` is the day of the month.
+        Second `*` is the month.
+        Third `*` is the day of the week starting with monday=0.
+        `*` is a wildcard that is always matched.
+        e.g.
+        `* * *` Will always return True.
+        `* * 0` Will only return True on Mondays.
+
+        Args:
+            frequency: str. A frequency string that is checked to see if a backup should occur today.
+        Returns:
+            True if today matches the frequency, False if it does not.
+        """
+        if not len(frequency) == 5:
+            logger.error(f"Frequency string is invalid - {frequency}")
+            return False
```

Comment:
My opinion is that it should raise an exception that is then handled a bit further up the stack to stop this strategy from continuing but not stopping any other strategies from running.

Reply:
Since this has changed into a method `should_run` which now lives on the remote strategy my opinion is that we should return False and log the error in the `should_run` method. This way we get the error message but the remote strategy does not run without breaking the flow.